crates/proposer-client/src/network_prover.rs

```diff
@@ -32,10 +32,10 @@ impl AggregationProver for NetworkProver {
     }
 }
 
-pub fn new_network_prover(endpoint: &str) -> anyhow::Result<NetworkProver> {
+pub fn new_network_prover(endpoint: &url::Url) -> anyhow::Result<NetworkProver> {
```

Comment:
It could be just that (as Url is implementing `AsRef<str>`)
```suggestion
pub fn new_network_prover<T: AsRef<str>>(endpoint: T) -> anyhow::Result<NetworkProver> {
```

And just `rpc_url(endpoint)`